### PR TITLE
config: track message ancestors for unknown fields

### PR DESCRIPTION
**Pull request by timonwong**
Signed-off-by: Tianpeng Wang <tpwang@alauda.io>

Commit Message: config: track message ancestors for unknown fields
Additional Description:
Risk Level: Low
Testing: unit test
Docs Changes: N/A
Release Notes: Required
Platform Specific Features: N/A

### CODE REVIEW

**Comment by yanavlasov** [comment]
Needs main merge

/wait

**Review by adisuissa — DISMISSED** [review]
Code changes LGTM.
@htuch this PR removes the context parameter of the ProtoVisitor. Is there a plan to use it for anything, or can this be removed?

**Comment by mattklein123** [comment]
Sorry this will need a main merge due to my recent changes and then I can take a look.

/wait

**Comment by RyanTheOptimist** [comment]
This is still waiting for a main merge.

/wait

File: source/common/protobuf/utility.cc [hunk 1]
```diff
@@ -373,7 +373,8 @@ class UnexpectedFieldProtoVisitor : public ProtobufMessage::ConstProtoVisitor {
                               Runtime::Loader* runtime)
       : validation_visitor_(validation_visitor), runtime_(runtime) {}
 
-  void onField(const Protobuf::Message& message, const Protobuf::FieldDescriptor& field) override {
+  void onField(const Protobuf::Message& message, const Protobuf::FieldDescriptor& field,
+               absl::Span<const Protobuf::Message* const>) override {
```
Comment (mattklein123) [inline source/common/protobuf/utility.cc:377]: Since this is not used right now maybe just skip passing parents in? We can always add this later.

**Review by mattklein123 — CHANGES_REQUESTED** [review]
Nice, this is a great improvement, thanks. Just some small comments. Thank you.

/wait

**Pull request merged**